addons/website_livechat/controllers/chatbot.py

```diff
@@ -13,6 +13,14 @@ def chatbot_test_script(self, chatbot_script):
         As we don't have a im_livechat.channel linked to it, we pre-emptively create a discuss.channel
         that will hold the conversation between the bot and the user testing the script. """
 
+        channels = request.env["discuss.channel"].search([
+            ["is_member", "=", True],
+            ["livechat_active", "=", True],
+            ["chatbot_current_step_id.chatbot_script_id", "=", chatbot_script.id],
+        ]).sudo()
+        for channel in channels:
+            channel.sudo()._close_livechat_session()
```

Comment:
comment why sudo is necessary and acceptable

Reply:
`sudo()`s removed. only user can access this route and it works fine.